src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -184,13 +184,15 @@ private static IExpression CreateParamArray(IParameterSymbol parameter, Immutabl
             if (parameter.Type.TypeKind == TypeKind.Array)
             {
                 IArrayTypeSymbol arrayType = (IArrayTypeSymbol)parameter.Type;
-                ArrayBuilder<IExpression> paramArrayArguments = ArrayBuilder<IExpression>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+                ArrayBuilder<IExpression> builder = ArrayBuilder<IExpression>.GetInstance(boundArguments.Length - firstArgumentElementIndex);
+
                 for (int index = firstArgumentElementIndex; index < boundArguments.Length; index++)
                 {
-                    paramArrayArguments.Add(boundArguments[index]);
+                    builder.Add(boundArguments[index]);
                 }
+                var paramArrayArguments = builder.ToImmutableAndFree();
 
-                return new ArrayCreation(arrayType, paramArrayArguments.ToImmutableAndFree(), boundArguments.Length - 1 > firstArgumentElementIndex ? boundArguments[firstArgumentElementIndex].Syntax : null);
+                return new ArrayCreation(arrayType, paramArrayArguments, paramArrayArguments.Length > 0 ? paramArrayArguments[0].Syntax : null);
```

Comment:
I tried to create a SynxtaxList instead of using he first params array element for it's Syntax, but it doesn't have a location. @srivatsn Is there a way to solve this? 
```cs
var argumentList = SyntaxFactory.List<SyntaxNode>(paramArrayArguments.Select(expression => expression.Syntax));
```